### PR TITLE
TNO-2598 Update Reports

### PR DESCRIPTION
**Pull request by Fosol**
Improved the workflow and UI to inform a user what will occur if they keep an active report instance and turn on the auto report functionality.  Also provided a way to control how it will work.

## Summary

- Publish tno-core:0.1.56
- Updated Editor app
- Updated Subscriber app
- Added configuration to control whether an auto report will generate a new instance or use an active one
- Fixed DevOps service account issue
- Fixed SignalR report update issue

## Subscriber Edit Report

image

image

### CODE REVIEW

File: app/subscriber/src/features/my-reports/edit/ReportEditPage.tsx [hunk 1]
```diff
@@ -150,7 +150,7 @@ export const ReportEditPage = () => {
     // Report has been updated, go fetch latest.
     // TODO: This can blow away a users' changes.
     try {
-      if (message.id === report.id) {
+      if (message.reportId === report.id) {
```
Comment (Fosol) [inline app/subscriber/src/features/my-reports/edit/ReportEditPage.tsx:153]: Fixed bug

File: openshift/kustomize/shared_resources/base/service-account.yaml [hunk 2]
```diff
@@ -0,0 +1,123 @@
+---
+apiVersion: v1
+kind: ServiceAccount
```
Comment (Fosol) [inline openshift/kustomize/shared_resources/base/service-account.yaml:3]: Fixed DevOps issue in DEV

File: app/editor/src/features/admin/reports/ReportFormSections.tsx [hunk 3]
```diff
@@ -146,10 +146,19 @@ export const ReportFormSections = () => {
...
+              checked={values.settings.content.copyPriorInstance}
+              onChange={(e) => {
+                setFieldValue('settings.content.copyPriorInstance', e.target.checked);
+              }}
+            />
+            <Checkbox
+              name={`settings.content.clearOnStartNewReport`}
               label="Empty report when starting next report"
-              checked={!values.settings.content.copyPriorInstance}
+              tooltip="If this is off an auto report will not generate a new report if an active one already exists."
```
Comment (kylgarmor) [inline app/editor/src/features/admin/reports/ReportFormSections.tsx:158]: "off" is a funny word here for me. I think "not set" might be better? Also, add a comma for after "off" ;)

File: app/subscriber/src/features/my-reports/edit/settings/ReportEditSendForm.tsx [hunk 4]
```diff
@@ -92,11 +92,37 @@ export const ReportEditSendForm = ({ onPublish, onGenerate }: IReportEditSendFor
             <p>
               An Auto Report is created for you on a schedule, but sent manually. If the report is
               not sent before the next scheduled run, it will <strong>not</strong> generate a new
-              report.
+              report unless you select 'Empty report when staring next report'.
```
Comment (kylgarmor) [inline app/subscriber/src/features/my-reports/edit/settings/ReportEditSendForm.tsx:95]: typo "staring" should be "starting"

File: app/subscriber/src/features/my-reports/edit/settings/template/ReportSchedule.tsx [hunk 5]
```diff
@@ -61,6 +61,13 @@ export const ReportSchedule: React.FC<IReportScheduleProps> = ({ index, label })
             }
           }}
         />
+
+        <Col className="info">
+          <p>
+            A schedule that has already populated a report in a given day, will not attempt to do so
```
Comment (kylgarmor) [inline app/subscriber/src/features/my-reports/edit/settings/template/ReportSchedule.tsx:67]: no comma needed here after "day"

**Pull request merged**